hathor/transaction/storage/transaction_storage.py

```diff
@@ -1019,7 +1019,7 @@ def _run_topological_sort_dfs(self, root: BaseTransaction, visited: Dict[bytes,
 
     def add_to_indexes(self, tx: BaseTransaction) -> None:
         if not self.with_index:
-            raise NotImplementedError
+            return  # XXX: avoid failing on some situations where this is called before we know it's OK to skip
```

Comment:
Maybe it will be difficult for some reading the code to know what situations are these.

Should we link to this PR's description, or create an issue to implement the builder pattern you suggested, describing the problem there, then link to that issue?

Reply:
I think linking to the PR makes sense and should be enough, I'll do that.

Reply:
Done.